### ISSUE TITLE
Failed to connect to the remote extension

### ISSUE THREAD

**Issue by arthurgeron** [microsoft/vscode-remote-release#1104]
Issue Type: <b>Bug</b>

Upon opening window on distro "Ubuntu" I get this error:
`Failed to connect to the remote extension host server (Error: connect ETIMEDOUT 172.22.239.58:46219)`   
At other times I get the same error but at the port 38489   

### Useful Info:   
I can ping 172.22.239.58 normally.   
Beforementioned is not accessible via SSH by default.   
I worked flawlessly in the past, without any configuration needed.   
SSH is accessible via localhost.   

### Solutions I've tried:   
- Reset /etc/resolv.conf
- Restart WSL
- Add ssh keys to authorized_keys
- Reinstall Ubuntu

Extension version: 0.39.1
VS Code version: Code - Insiders 1.37.0-insider (4a03f7e7eff2560d2e53c029321870514d19a2aa, 2019-08-05T09:34:10.366Z)
OS version: Windows_NT x64 10.0.18950

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HK CPU @ 2.70GHz (8 x 2712)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|11.87GB (1.54GB free)|
|Process Argv||
|Screen Reader|no|
|VM|22%|
</details>

**Comment by roblourens** [comment]
Please share the log from the Remote-SSH output channel

**Comment by arthurgeron** [comment]
The log bellow is what I was able to find under the Window output channel, I couldn't find a Remote-SSH output channel as the image shows:
![image](https://user-images.githubusercontent.com/3487334/62902563-5eed3b80-bd36-11e9-8f60-3c58008aa375.png)


```
[2019-08-12 19:19:03.386] [renderer3] [error] connect ETIMEDOUT 172.22.137.11:38197: Error: connect ETIMEDOUT 172.22.137.11:38197
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1113:14)
[2019-08-12 19:19:03.387] [renderer3] [error] Cannot read property 'extensions' of null: TypeError: Cannot read property 'extensions' of null
    at P.<anonymous> (file:///C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:4034:141)
    at Generator.next (<anonymous>)
    at r (file:///C:/Program Files/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:35:293)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
```

**Comment by roblourens** [comment]
It looks like you are on a rather old version of the extension, could you try updating?

**Comment by arthurgeron** [comment]
It doesn't show a option to update for me, unfortunately:
![image](https://user-images.githubusercontent.com/3487334/62906222-48e67780-bd44-11e9-826f-6bc6fdb1f6eb.png)

**Comment by roblourens** [comment]
Are you trying to use SSH or WSL? You confused me by mentioning SSH a few times. Please give more details about what you are doing when you say "Upon opening window"

**Comment by arthurgeron** [comment]
What I meant was that I'm clicking on the Remote - WSL icon in the bottom left corner of the visual code insiders window, then clicking on `Remote-WSL: New Window using Distro...`. I mentioned SSH because I read that this extension uses SSH to connect to the Distros

**Comment by roblourens** [comment]
Ok, sorry I misunderstood

**Comment by aeschli** [comment]
Can you invoke the `Remote-WSL: Show Log...` command and paste what you find in the WSL log (it's part of the terminal view)

**Comment by arthurgeron** [comment]
I only have that command in a Regular Window, the Visual Code's WSL  Window does not have that option.
![image](https://user-images.githubusercontent.com/3487334/63267204-e8859780-c267-11e9-9223-b38122bc4131.png)

**Comment by arthurgeron** [comment]
Also, logs on a normal window shows nothing.

**Comment by arthurgeron** [comment]
If you want to see the issue yourself via screen share or something let me know.

**Comment by arthurgeron** [comment]
This is what was printed in the developer tools console:
![image](https://user-images.githubusercontent.com/3487334/63268748-423b9100-c26b-11e9-90e7-dd79d0e73621.png)

**Comment by aeschli** [comment]
Hm, strange. Can you go to the Terminal view and check if there's a WSL instance?

**Comment by arthurgeron** [comment]
Nope, looks like since it doesn't show terminal view since didn't finish connecting:
![image](https://user-images.githubusercontent.com/3487334/63436426-0f2a0680-c3ff-11e9-87dd-2701e237a38b.png)

**Comment by arthurgeron** [comment]
Also output channels show nothing about WSL
![image](https://user-images.githubusercontent.com/3487334/63436504-308af280-c3ff-11e9-8919-fffd1d308953.png)

**Comment by arthurgeron** [comment]
I've seen issue #914  and it looks to be somewhat related.. I looked into the path, and noticed that when I try to launch code-insiders from within wsl it hangs, interesting.

**Comment by arthurgeron** [comment]
I think I'll probably need to wait until [this bug](https://github.com/microsoft/WSL/issues/4337) is fixed

**Comment by aeschli** [comment]
When the remote window comes up, you see a progress notification (Starting Code in WSL ...). The first part is a link, can you click on it? 

Are are on WSL2?

**Comment by arthurgeron** [comment]
I'll try clicking on it once I get home and yes I'm in WSL 2

**Comment by arthurgeron** [comment]
Clicking on it broght me to the terminal view of WSL.
Here's the output log:
```
[2019-08-22 12:15:15.891] Starting VS Code Server inside WSL (Ubuntu)
[2019-08-22 12:15:15.891] Extension version: 0.39.3, Windows build: 18963. Multi distro support: enabled. WSL path support: enabled
[2019-08-22 12:15:16.081] Probing if server is already installed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server-insiders/bin/ada79d72be36ecbfed6f88bb16edcc5cc3f5c2a8 ] && echo found || echo missing"
[2019-08-22 12:15:16.240] No server install found on WSL, downloading server on client side: C:\Users\arthu\AppData\Local\Temp\vscode-remote-wsl\ada79d72be36ecbfed6f88bb16edcc5cc3f5c2a8\vscode-server-linux-x64.tar.gz
[2019-08-22 12:15:19.842] Launching C:\WINDOWS\System32\wsl.exe -d Ubuntu sh -c "\"\$(wslpath -u 'c:/Users/arthu/.vscode-insiders/extensions/ms-vscode-remote.remote-wsl-0.39.3/scripts/wslServer.sh')\" 'ada79d72be36ecbfed6f88bb16edcc5cc3f5c2a8' 'insider' '.vscode-server-insiders' 'C:/Users/arthu/AppData/Local/Temp/vscode-remote-wsl/ada79d72be36ecbfed6f88bb16edcc5cc3f5c2a8/vscode-server-linux-x64.tar.gz' 0  --disable-telemetry" in c:\Users\arthu\.vscode-insiders\extensions\ms-vscode-remote.remote-wsl-0.39.3
[2019-08-22 12:15:20.006] WSL version: 4.19.57-microsoft-standard Ubuntu
[2019-08-22 12:15:20.006] Updating server...
[2019-08-22 12:15:20.006] Updating VS Code Server to version ada79d72be36ecbfed6f88bb16edcc5cc3f5c2a8
[2019-08-22 12:15:20.006] Removing previous installation...
[2019-08-22 12:15:20.006] 100%
[2019-08-22 12:15:20.006] Unpacking:   0%
[2019-08-22 12:15:21.2 23%
[2019-08-22 12:15:21.5 68%
[2019-08-22 12:15:21.8100%
[2019-08-22 12:15:22.145] 
[2019-08-22 12:15:22.146] Starting server: /home/itsme/.vscode-server-insiders/bin/ada79d72be36ecbfed6f88bb16edcc5cc3f5c2a8/server.sh  --port=0 --enable-remote-auto-shutdown --disable-telemetry
[2019-08-22 12:15:22.446] 
[2019-08-22 12:15:22.446] 
[2019-08-22 12:15:22.446] *
[2019-08-22 12:15:22.446] * Visual Studio Code Server
[2019-08-22 12:15:22.446] *
[2019-08-22 12:15:22.446] * Reminder: You may only use this software with Visual Studio family products,
[2019-08-22 12:15:22.446] * as described in the license https://aka.ms/vscode-remote/license
[2019-08-22 12:15:22.446] *
[2019-08-22 12:15:22.446] 
[2019-08-22 12:15:22.446] 
[2019-08-22 12:15:22.446] IP Address: 172.21.229.31
[2019-08-22 12:15:22.446] webview server listening on 42337
[2019-08-22 12:15:22.446] Extension host agent listening on 43679
[2019-08-22 12:15:22.454] WSL resolver response: 172.21.229.31:43679
```

**Comment by aeschli** [comment]
Thanks a lot. It looks like the VSCode server starts properly. But I assume after that you get the message that VSCode can't connect to 172.21.229.31:43679.

You can open a local browser and open http://172.21.229.31:43679/version to see if it is reachable.

**Comment by arthurgeron** [comment]
No problem! Thank you for the info. 
When I try to reach the resolver response address under the http protocol I get a timeout reponse in the browser.
PS: In this example the resolver address was `WSL resolver response: 172.21.232.124:44247`:
![image](https://user-images.githubusercontent.com/3487334/63521137-431f2d80-c4cc-11e9-9a60-3ebb23cadfd3.png)

**Comment by arthurgeron** [comment]
if I do a `netstat -tulpn` on Ubuntu's bash I can see that the server is indeed running, just isn't reachable:
![image](https://user-images.githubusercontent.com/3487334/63521251-75c92600-c4cc-11e9-95af-78022f65e641.png)

**Comment by arthurgeron** [comment]
But it is reachable under localhost (since WSL 2 and windows share the same localhost), I wonder if I could set up the extension to use localhost instead.
![image](https://user-images.githubusercontent.com/3487334/63521321-998c6c00-c4cc-11e9-93bd-1f98a0cd89ff.png)

**Comment by aeschli** [comment]
WSL2 is running in it's own VM, thus with it's own IP. I don't think the WSL team found a solution around that.

**Comment by aeschli** [comment]
Please add the `/version` path: 'localhost:44247/version'. You will see that it's not there.

**Comment by arthurgeron** [comment]
In the past WSL 2 didn't share localhost with the windows host, I've noticed that after they implemented the shared localhost trying to access a WSL 2 distro via it's ip address stopped working.

**Comment by arthurgeron** [comment]
see [this link](https://blogs.windows.com/windowsexperience/2019/07/26/announcing-windows-10-insider-preview-build-18945/) for reference

**Comment by arthurgeron** [comment]
Is it possible that we get an option to make the extension use localhost instead?

**Comment by aeschli** [comment]
Eventually we can change to always use localhost.
I wonder why connecting to the ip doesn't work for you. I have the same install (10.0.18950) and it connecting to the IP works.

**Comment by arthurgeron** [comment]
I have no idea either. I had tried everything, from re-installing WSL to switch to WSL 1 and back to 2.. Since I’m on Windows Preview it could be really anything, I’ve given up on WSL for now since it was eating up all my RAM. Let me know when it works with localhost so I can test it again!

Thank you.

________________________________

**Comment by costas90** [comment]
#1304 I have a similar issue.

I noticed that my Antivirus firewall is causing this, because it works when I turn it off. Also happened after updating windows to v. 1903.

@arthurgeron is this the same for you also? I am not sure what to change in my Firewall settings to make this work.

**Comment by arthurgeron** [comment]
@costas90 I don't use a custom anti-virus or firewall, I've tried disabling windows firewall to no avail.

**Comment by arthurgeron** [comment]
Windows build 18970 fixed interop bug, which fixed the issue mentioned here, hooray!

**Comment by arthurgeron** [comment]
When I re-enabled WSL windows set my default WSL version to 1.. so  actually it's still not working in WSL 2

**Comment by arthurgeron** [comment]
is there any ETA for this extension to support localhost connections?

**Comment by mdmnk** [comment]
Same issue on this end `Unable to open '_<FILENAME>_': Unable to read file (Error: File not found (\__vscode-remote-uri__\home\_PATH_TO_FILE_)).`

**Comment by aeschli** [comment]
Fixed in 0.39.7 (requires latest insider build).

For WSL2 (> Windows Build 18970) we now connect to localhost (::1)
